### DISCUSSION TITLE
How to send query to Cloud SQL for PostgreSQL

### DISCUSSION THREAD

**Discussion by uS-aito** [kedacore/keda#1662] [Q&A / Need Help] [answered]
Hello, 

We are using Cloud SQL for PostgreSQL and would like to do some scaling based on the number of data in a table.

Normally, the connection to Cloud SQL is made by injecting a SQL Proxy as a sidecar and connecting to localhost (127.0.01).

How can we achieve this in KEDA?
Do I just add the sidecar to the Deployment to be scaled? Or do I need to add a sidecar to the keda-operator-metrics-apiserver Pod by some means?

Thank you.

**Comment by zroubalik** [comment, marked as answer]
Hi, I am not that much familiar with Cloud SQL, but IMHO it **should** work if you add the sidecar to the both KEDA Operator and Metrics Apiserver (you need to add it to the both Deployments).
I am curious to hear the result :)

**Reply by uS-aito** [reply]
Thanks for replying.

After reading your reply and reading the source code of KEDA, I came to the same conclusion.
In other words, I thought that I need to inject sidecar into both pods.
I just tried scaling after injecting sidecar into both pods, and it worked fine.

Thanks for your help.

**Reply by kokizzu** [reply]
what sidecar @uS-aito ? 
cloudsql postgres -> metricserver -> keda? because [keda-postgres](https://keda.sh/docs/2.11/scalers/postgresql/) cannot have cloudsql's certificate injected?

**Reply by kokizzu** [reply]
or the sidecar is to proxy requests between keda and cloudsql, so the cloudsql certs is on the sidecar?

**Comment by uS-aito** [comment]
I'm not familiar with github discussion, how do I mark this topic as answerd?

**Reply by zroubalik** [reply]
I did that :)

**Reply by uS-aito** [reply]
Thank you very much!

**Comment by Joshua-Hager** [comment]
Checking in 4 years later to let people know that this solution still works. The following steps worked for me:

- Download and modify the [release YAML file](https://github.com/kedacore/keda/releases/download/v2.18.1/keda-2.18.1.yaml) with the [Cloud SQL Proxy Sidecar](https://cloud.google.com/sql/docs/mysql/connect-kubernetes-engine#proxy-sidecar-pattern) for the Operator container and the Metrics API Server container.
- Apply the modified YAML file to your cluster. You may need to use the `--server-side` option when applying due to the length of the file i.e. `kubectl apply --server-side -f /path/to/keda.yml`.
- When writing the MySQL scaler, use the [password authentication method](https://keda.sh/docs/2.18/scalers/mysql/#authentication-parameters) and pass the parameters individually (the connection string method resulted in a TCP dial error for me).
- Bind the keda-operator service account to a GCP service account with the correct IAM permissions to access your Cloud SQL instance or change the name of the of the keda-operator service account to the name of a service account that already has the correct permissions.

These steps allowed the MySQL scaler to query the Cloud SQL instance.
